TripletCriterion.cu

```diff
@@ -7,21 +7,35 @@
 __global__ void triplet_dist_kernel(const int n, const float norm,
                                     const int nb_batch, const int length,
                                     float* x, float* y) {
+
+  // Unnecessary for loop...
+  // This is EXACTLY equivalent to `if (i < n)` statement
   for (int i = blockIdx.x*blockDim.x+threadIdx.x; i < n; i += blockDim.x*gridDim.x) {
```

Comment:
Hmm... you set `n = blockDim.x*gridDim.x`, so it's hard to have `i += n`, no?
Moreover, I believe these are logical *grid* and *block*, and do not have a 1-1 hardware correspondence. I think that *block*s are made of nearby cores, so that the *thread*s can easily share data.